core/trino-main/src/main/java/io/trino/metadata/FunctionInvoker.java

```diff
@@ -55,4 +48,41 @@ public List<Class<?>> getLambdaInterfaces()
     {
         return lambdaInterfaces;
     }
+
+    public static Builder builder()
+    {
+        return new Builder();
+    }
+
+    public static class Builder
+    {
+        private MethodHandle methodHandle;
+        private MethodHandle instanceFactory;
+        private List<Class<?>> lambdaInterfaces = ImmutableList.of();
+
+        private Builder() {}
+
+        public Builder setMethodHandle(MethodHandle methodHandle)
```

Comment:
`withMethodHandle(...)` or just `methodHandle(...)`? Same for the ones below. BTW, in the commit to add a builder for AggregationImplementation you use the latter convention.